qt/applications/workbench/workbench/plotting/figurewindow.py

```diff
@@ -118,3 +132,11 @@ def _plot_on_here(self, names, ax):
         else:
             plot_from_names(names, errors=(fig_type == FigureType.Errorbar),
                             overplot=ax, fig=fig)
+
+    def _on_focusWindowChanged(self, window):
+        """
+        The figure window should always remain on top of the main
+        Workbench window.
+        """
+        if window and MAIN_WINDOW_OBJECT_NAME + 'Window' == window.objectName():
```

Comment:
Is the object name of the mainwindow guranteed to be this? Would it be better to use `get_main_window_widget` and test `isinstance()`?

Reply:
This was discussed offline and more detailed comments have been added to explain why this comparison is used.